sway-core/src/semantic_analysis/ast_node/declaration/auto_impl.rs

```diff
@@ -520,9 +521,20 @@ where
         let mut reads = false;
         let mut writes = false;
 
+        // used to check for name collisions
+        let mut names = BTreeMap::new();
+
+        // generate code
         for r in contract_fns {
             let decl = engines.de().get(r);
 
+            // Contract methods must be unique
+            if let Some(old) = names.insert(decl.name.as_str().to_string(), decl.name.span()) {
+                return Err(handler.emit_err(
+                    CompileError::MultipleContractsMethodsWithTheSameName { span: old },
```

Comment:
In rust, the compiler highlights one of the duplicate definitions and the error message indicates the location of the other definition. We should consider doing something similar so it's clear where to find the other definition.

<img width="788" alt="image" src="https://github.com/user-attachments/assets/92323840-cfdf-4c44-a29f-0beeb55ce5cc" />